cluster_pack/uploader.py

```diff
@@ -309,11 +340,29 @@ def _upload_env_from_venv(
         if not resolved_fs.exists(dir):
             resolved_fs.mkdir(dir)
         _logger.info(f'Uploading env at {local_package_path} to {package_path}')
-        resolved_fs.put(local_package_path, package_path)
+
+        if not _try_put_atomic(resolved_fs,
+                               local_package_path,
+                               package_path,
+                               lambda path: _is_archive_up_to_date(path, reqs, resolved_fs)):

```

Comment:
Passing a function to a function is not super pythonic IMO + it seems the exact same function is passed on the second call you do to the _try_put_atomic function coudln't we integrate it to the _try_put_atomic function itself ?

Reply:
See comment above, will simplify, especially since now there is only one way to verify the package (as we no longer affect the code path from a venv, only from a pex)

Reply:
To clarify: this commented change will disappear.